tools/devops/automation/templates/build/build.yml

```diff
@@ -52,7 +52,13 @@ steps:
 
 - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
   - pwsh: |
-      git reset --hard HEAD^2 
+      git config remote.origin.fetch '+refs/pull/*:refs/remotes/origin/pull/*'
```

Comment:
Add all the refs for the next fetch.

Reply:
This can be quite a few references, you might want to limit to just the PR we're checking out (makes fetching faster):

> git config remote.origin.fetch '+refs/pull/<number>/*:refs/remotes/origin/pull/<number>/*'

Reply:
Yep, I'll land this and will move the code to a template to be reused.